### PR TITLE
Abacus Keyboard ReMerge

### PR DESCRIPTION
**Pull request by nickolaij**
## Description

I've pulled down my last merge request, made all the changes and some more to make it more professional and re-opened it. Thanks for the help already, just let me know what else I need to do :)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/abacus/keymaps/default/readme.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# The default keymap for abacus
```
Comment (noroadsleft) [inline keyboards/abacus/keymaps/default/readme.md]:
```suggestion
# The default keymap for Abacus
```

File: keyboards/abacus/readme.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# abacus
```
Comment (noroadsleft) [inline keyboards/abacus/readme.md]:
```suggestion
# Abacus
```

File: keyboards/abacus/readme.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+![abacus](https://i.imgur.com/IFtuWaK.jpg)
+
+A first attempt at a PCB design for a mechanical keyboard. Includes encoder and RGB
```
Comment (noroadsleft) [inline keyboards/abacus/readme.md]:
```suggestion
A first attempt at a PCB design for a mechanical keyboard. Includes rotary encoder and RGB underglow.
```

File: keyboards/abacus/readme.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+* PCB Files: [Abacus PCB Github](https://github.com/nickolaij/Abacus_Rev2)
+* Keyboard Maintainer: [nickolaij](https://github.com/nickolaij)
+* Hardware Supported: [Elite C Microcontroller](https://keeb.io/products/elite-c-usb-c-pro-micro-replacement-arduino-compatible-atmega32u4) or Pro Micro Microcontroller (Elite C has additional pins for encoder)
+* Hardware Availability: https://github.com/nickolaij
```
Comment (noroadsleft) [inline keyboards/abacus/readme.md]: Slight rework here.

```suggestion
* Keyboard Maintainer: [nickolaij](https://github.com/nickolaij)
* Hardware Supported: Abacus PCB, [Elite C Microcontroller](https://keeb.io/products/elite-c-usb-c-pro-micro-replacement-arduino-compatible-atmega32u4) or Pro Micro Microcontroller (Elite C has additional pins for encoder)
* Hardware Availability: [Abacus PCB Github](https://github.com/nickolaij/Abacus_Rev2)
```

File: keyboards/abacus/rules.mk [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
```
Comment (noroadsleft) [inline keyboards/abacus/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = no        # Mouse keys
```

File: keyboards/abacus/info.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
+{
+    "keyboard_name": "Abacus",
+    "url": "https://www.github.com/nickolaij",
+    "maintainer": "nickolaij",
+    "width": 12,
```
Comment (noroadsleft) [inline keyboards/abacus/info.json]:
```suggestion
    "width": 12.75,
```

File: keyboards/abacus/info.json [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+                {"label":"k12", "x":2, "y":1, "w":1},
+                {"label":"k13", "x":3, "y":1, "w":1},
+                {"label":"k14", "x":4, "y":1, "w":1},
+                {"label":"k15", "x":5, "y":1, "w":1},
+                {"label":"k16", "x":6, "y":1, "w":1},
+                {"label":"k17", "x":7, "y":1, "w":1},
+                {"label":"k18", "x":8, "y":1, "w":1},
+                {"label":"k19", "x":9, "y":1, "w":1},
+                {"label":"k1a", "x":10, "y":1, "w":1},
+                {"label":"k1b", "x":11, "y":1, "w":1.5},
```
Comment (noroadsleft) [inline keyboards/abacus/info.json]:
```suggestion
                {"label":"k11", "x":1.25, "y":1, "w":1},
                {"label":"k12", "x":2.25, "y":1, "w":1},
                {"label":"k13", "x":3.25, "y":1, "w":1},
                {"label":"k14", "x":4.25, "y":1, "w":1},
                {"label":"k15", "x":5.25, "y":1, "w":1},
                {"label":"k16", "x":6.25, "y":1, "w":1},
                {"label":"k17", "x":7.25, "y":1, "w":1},
                {"label":"k18", "x":8.25, "y":1, "w":1},
                {"label":"k19", "x":9.25, "y":1, "w":1},
                {"label":"k1a", "x":10.25, "y":1, "w":1},
                {"label":"k1b", "x":11.25, "y":1, "w":1.5},
```

File: keyboards/abacus/info.json [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+                {"label":"k22", "x":2, "y":2, "w":1},
+                {"label":"k23", "x":3, "y":2, "w":1},
+                {"label":"k24", "x":4, "y":2, "w":1},
+                {"label":"k25", "x":5, "y":2, "w":1},
+                {"label":"k26", "x":6, "y":2, "w":1},
+                {"label":"k27", "x":7, "y":2, "w":1},
+                {"label":"k28", "x":8, "y":2, "w":1},
+                {"label":"k29", "x":9, "y":2, "w":1},
+                {"label":"k2a", "x":10, "y":2, "w":1},
+                {"label":"k2b", "x":11, "y":2, "w":1.25},
```
Comment (noroadsleft) [inline keyboards/abacus/info.json]:
```suggestion
                {"label":"k21", "x":1.75, "y":2, "w":1},
                {"label":"k22", "x":2.75, "y":2, "w":1},
                {"label":"k23", "x":3.75, "y":2, "w":1},
                {"label":"k24", "x":4.75, "y":2, "w":1},
                {"label":"k25", "x":5.75, "y":2, "w":1},
                {"label":"k26", "x":6.75, "y":2, "w":1},
                {"label":"k27", "x":7.75, "y":2, "w":1},
                {"label":"k28", "x":8.75, "y":2, "w":1},
                {"label":"k29", "x":9.75, "y":2, "w":1},
                {"label":"k2a", "x":10.75, "y":2, "w":1},
                {"label":"k2b", "x":11.75, "y":2, "w":1},
```

File: keyboards/abacus/info.json [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+                {"label":"k30", "x":0, "y":3, "w":1.25},
+                {"label":"k31", "x":1, "y":3, "w":1},
+                {"label":"k32", "x":2, "y":3, "w":1},
+                {"label":"k33", "x":3, "y":3, "w":2.75},
+                {"label":"k34", "x":4, "y":3, "w":2.75},
+                {"label":"k35", "x":5, "y":3, "w":1},
+                {"label":"k36", "x":6, "y":3, "w":1},
+                {"label":"k37", "x":7, "y":3, "w":1},
+                {"label":"k38", "x":8, "y":3, "w":1}
+
```
Comment (noroadsleft) [inline keyboards/abacus/info.json]:
```suggestion
                {"label":"k31", "x":1.25, "y":3, "w":1},
                {"label":"k32", "x":2.25, "y":3, "w":1},
                {"label":"k33", "x":3.25, "y":3, "w":2.75},
                {"label":"k34", "x":6, "y":3, "w":2.75},
                {"label":"k35", "x":8.75, "y":3, "w":1},
                {"label":"k36", "x":9.75, "y":3, "w":1},
                {"label":"k37", "x":10.75, "y":3, "w":1},
                {"label":"k38", "x":11.75, "y":3, "w":1}
```

**Comment by noroadsleft** [comment]
Still two unresolved issues on my previous review:

- https://github.com/qmk/qmk_firmware/pull/8308#discussion_r388473088
- https://github.com/qmk/qmk_firmware/pull/8308#discussion_r388478776

**Comment by nickolaij** [comment]
my bad, are fixed now

File: keyboards/abacus/rules.mk [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
```
[the remark this answers is not in the document]
Comment (noroadsleft) [reply keyboards/abacus/rules.mk]: Actually, my understanding is that Elite C has atmel-dfu from the factory.

File: keyboards/abacus/rules.mk [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = qmk-dfu
```
Comment (Duckle29) [inline keyboards/abacus/rules.mk]:
```suggestion
BOOTLOADER = atmel-dfu
```
Sorry. As @noroadsleft pointed out, I was wrong about Elite-C, and atmel-dfu was correct. I apologize for the trouble.

**Comment by tzarc** [comment]
I can see you're using your own local `master` branch -- can I also suggest going through this documentation:
https://docs.qmk.fm/#/newbs_git_using_your_master_branch

It'll help you keep up to date in future.

**Comment by nickolaij** [comment]
@tzarc Ah I see, definitely will do

**Comment by noroadsleft** [comment]
Thanks!

[This doc](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) should sort you through fixing your branches. If you need any help just ask!

**Pull request merged**